pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -328,6 +329,16 @@ public void removeProducer(Producer producer) {
         return delete(false, false, false);
     }
 
+    @Override
+    public CompletableFuture<MessageId> terminate() {
+        CompletableFuture<MessageId> future = new CompletableFuture<>();
+        producers.values().forEach(Producer::disconnect);
```

Comment:
why not disconnect sub?

Reply:
do we need retain existing consumer when non-persistent topic terminate? @gaoran10 @codelipenghui /cc

Reply:
I think non-persistent termination should behave like persistent